### PR TITLE
Add CDI default storage class alert runbooks

### PR DESCRIPTION
**Pull request by arnongilboa**
CDINoDefaultStorageClass & CDIDefaultStorageClassDegraded.

### CODE REVIEW

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# CDIDefaultStorageClassDegraded
+<!-- Edited by agilboa, 4 Dec 2023 -->
+
+## Meaning
+
+This alert fires when the default (Kubernetes or virtualization) storage class supports smart clone (either CSI or snapshot based) and ReadWriteMany.
```
Comment (sradco) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:6]: @arnongilboa maybe a link about smart clone would be helpful. Like https://github.com/kubevirt/containerized-data-importer/blob/main/doc/smart-clone.md

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+Get the default virtualization storage class by:
+```bash
+$ kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name'
+```
+
+Check the default storage class supports ReadWriteMany:
+```bash
+$ kubectl get storageprofile <default-storage-class-name> -o json | jq '.status.claimPropertySets'| grep ReadWriteMany
```
Comment (sradco) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:30]: @arnongilboa Is the <default-storage-class-name>  the "virtualization storage class" or "Kubernetes storage class"?

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```bash
+$ kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubernetes.io/is-default-class"=="true")|.name'
+```
+
+Get the default virtualization storage class by:
+```bash
+$ kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name'
+```
+
+Check the default storage class supports ReadWriteMany:
```
[the remark this answers is not in the document]
Comment (sradco) [reply docs/runbooks/CDIDefaultStorageClassDegraded.md:28]: @arnongilboa  If you mean that you need to check for both I would do :

1. export CDI_KUBERNETES_SC="$(kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubernetes.io/is-default-class"=="true")|.name')
2. $ kubectl get storageprofile $CDI_KUBERNETES_SC -o json | jq '.status.claimPropertySets'| grep ReadWriteMany
3. export CDI_VIRTUALIZATION_SC="$(kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name')
4. $ kubectl get storageprofile $CDI_VIRTUALIZATION_SC -o json | jq '.status.claimPropertySets'| grep ReadWriteMany

WDYT?

File: docs/runbooks/CDINoDefaultStorageClass.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+Get the default virtualization storage class by:
+
+```bash
+$ kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name'
+```
+
+## Mitigation
+
+Ensure that there is one storage class that has the default (k8s or virt) storage class annotation.
```
Comment (sradco) [inline docs/runbooks/CDINoDefaultStorageClass.md:30]: @arnongilboa  Is there a documentation for setting a default storage class?

File: docs/runbooks/CDINoDefaultStorageClass.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+<!--DS: If you cannot resolve the issue, log in to the link:https://access.redhat.com[Customer Portal] and open a support case, attaching the artifacts gathered during the Diagnosis procedure.-->
```
Comment (sradco) [inline docs/runbooks/CDINoDefaultStorageClass.md:32]: @arnongilboa no need for the comments

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+<!--USstart-->
+See [doc](https://github.com/kubevirt/containerized-data-importer/blob/main/doc/efficient-cloning.md) for details about smart clone prerequisites.
+<!--USend-->
+
+## Mitigation
+
+Ensure that the default storage class supports smart clone  and ReadWriteMany.
+
+<!--DS: If you cannot resolve the issue, log in to the link:https://access.redhat.com[Customer Portal] and open a support case, attaching the artifacts gathered during the Diagnosis procedure.-->
```
Comment (sradco) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:41]: @arnongilboa no need for the comments

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+```bash
+$ kubectl get sc -o json | jq '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name'
+```
+
+Check the default storage class supports ReadWriteMany:
+```bash
+$ kubectl get storageprofile <default-storage-class-name> -o json | jq '.status.claimPropertySets'| grep ReadWriteMany
+```
+
+<!--USstart-->
```
Comment (sradco) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:33]: @arnongilboa no need for the comments

File: docs/runbooks/CDIDefaultStorageClassDegraded.md (hunk 2 above)
Comment (arnongilboa) [reply docs/runbooks/CDIDefaultStorageClassDegraded.md:30]: fixed

File: docs/runbooks/CDIDefaultStorageClassDegraded.md (hunk 1 above)
Comment (arnongilboa) [reply docs/runbooks/CDIDefaultStorageClassDegraded.md:6]: The link is in the Diagnosis section.

File: docs/runbooks/CDINoDefaultStorageClass.md (hunk 4 above)
Comment (arnongilboa) [reply docs/runbooks/CDINoDefaultStorageClass.md:30]: Added

**Comment by sradco** [comment]
/lgtm

@apinnick can you please review this?

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+If the default storage class does not support smart clone, we fallback to host-assisted cloning, which is the least efficient method of cloning.
+
+If the default storage class does not suppprt ReadWriteMany, a virtual machine using it is not live-migratable.
+
+## Diagnosis
+
+Get the default virtualization storage class:
+```bash
+$ export CDI_DEFAULT_VIRT_SC="$(kubectl get sc -o json | jq -r '.items[].metadata|select(.annotations."storageclass.kubevirt.io/is-default-virt-class"=="true")|.name')"
+$ echo defult_virt_sc=$CDI_DEFAULT_VIRT_SC
```
Comment (apinnick) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:21]: typo? "defult"

Comment (arnongilboa) [reply docs/runbooks/CDIDefaultStorageClassDegraded.md:21]: fixed

**Review by sradco — APPROVED** [review]

File: docs/runbooks/CDIDefaultStorageClassDegraded.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# CDIDefaultStorageClassDegraded
+<!-- Edited by agilboa, 6 Dec 2023 -->
+
+## Meaning
+
+This alert fires when the default (Kubernetes or virtualization) storage class supports smart clone (either CSI or snapshot based) and ReadWriteMany.
```
Comment (apinnick) [inline docs/runbooks/CDIDefaultStorageClassDegraded.md:6]: Is this first sentence correct? It sounds like something is missing. Should it be "This alert fires when the default storage class does NOT support smart clone and RWX"?

Comment (arnongilboa) [reply docs/runbooks/CDIDefaultStorageClassDegraded.md:6]: Right... Fixing in a follow-up.

**Pull request merged**